### PR TITLE
Replace expected Junit 4 @Test field with assertThatThrownBy

### PR DESCRIPTION
**Pull request by diega**
## PR description
To simplify a future Junit5 migration this PR removes the usage of the [expected](https://junit.org/junit4/javadoc/4.13/org/junit/Test.html#expected()) field of the `org.junit.Test` annotation with the AssertJ's `assertThatThrownBy` method.

- [HEAD~3](https://github.com/diega/besu/commit/expected_replace~3): Fix two tests from `o.h.b.e.vm.OperandStackTest`, the name of the tests referred to testing `o.h.b.e.internal.OperandStack#set` but they were actually testing _get_ (duplicated indeed, b/c proper tests for _get_ existed with the right method name) -> Thanks @macfarla!
- [HEAD~2](https://github.com/diega/besu/commit/expected_replace~2): Remove unreachable asserts
- [HEAD~1](https://github.com/diega/besu/commit/expected_replace~1): Syntactic wrapping of lines throwing the expected exception
- [HEAD](https://github.com/diega/besu/commit/expected_replace): Extract variables to let only a single method call inside the `assertThatThrownBy` lambda

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java [hunk 1]
```diff
@@ -83,17 +85,17 @@ public void get() {
     assertThat(stack.get(2)).isEqualTo(Bytes32.fromHexString("0x01"));
   }
 
-  @Test(expected = UnderflowException.class)
+  @Test
   public void set_NegativeOffset() {
     final OperandStack stack = new OperandStack(1);
-    stack.get(-1);
+    assertThatThrownBy(() -> stack.get(-1)).isInstanceOf(UnderflowException.class);
```
Comment (macfarla) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java:91]: we have two tests doing the same thing - should this one be stack.set()
see get_NegativeOffset

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java [hunk 2]
```diff
@@ -83,17 +85,17 @@ public void get() {
...
   }
 
-  @Test(expected = UnderflowException.class)
+  @Test
   public void set_IndexGreaterThanSize() {
     final OperandStack stack = new OperandStack(1);
     stack.push(UInt256.fromHexString("0x01"));
-    stack.get(2);
+    assertThatThrownBy(() -> stack.get(2)).isInstanceOf(UnderflowException.class);
```
Comment (macfarla) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java:98]: I think same here see get_IndexGreaterThanSize

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java [hunk 3]
```diff
@@ -65,41 +66,61 @@ public void createP2PNetwork() throws IOException {
...
         NetworkingConfiguration.create()
             .setDiscovery(DiscoveryConfiguration.create().setBindHost(null));
-    try (final P2PNetwork broken = builder().config(config).build()) {
-      Assertions.fail("Expected Exception");
-    }
+    final DefaultP2PNetwork.Builder p2pNetworkBuilder = builder().config(config);
+    assertThatThrownBy(
+            () -> {
+              try (final P2PNetwork ignored = p2pNetworkBuilder.build()) {
+                Assertions.fail("Expected Exception");
```
Comment (macfarla) [inline ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java:78]: do we still need this with assertThatThrownBy?

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java [hunk 4]
```diff
@@ -65,41 +66,61 @@ public void createP2PNetwork() throws IOException {
...
         NetworkingConfiguration.create()
             .setDiscovery(DiscoveryConfiguration.create().setBindHost("fake.fake.fake"));
-    try (final P2PNetwork broken = builder().config(config).build()) {
-      Assertions.fail("Expected Exception");
-    }
+    final DefaultP2PNetwork.Builder p2pNetworkBuilder = builder().config(config);
+    assertThatThrownBy(
+            () -> {
+              try (final P2PNetwork ignored = p2pNetworkBuilder.build()) {
+                Assertions.fail("Expected Exception");
```
Comment (macfarla) [inline ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java:93]: as above

**Review by macfarla — COMMENTED** [review]
overall LGTM. couple of comments

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java [hunk 5]
```diff
@@ -65,41 +66,61 @@ public void createP2PNetwork() throws IOException {
...
+        .isInstanceOf(IllegalArgumentException.class);
   }
 
-  @Test(expected = NullPointerException.class)
+  @Test
   public void createP2PNetwork_NullKeyPair() throws IOException {
-    try (final P2PNetwork broken = builder().config(config).nodeKey(null).build()) {
-      Assertions.fail("Expected Exception");
-    }
+    final DefaultP2PNetwork.Builder p2pNetworkBuilder = builder().config(config).nodeKey(null);
```
Comment (macfarla) [inline ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java:116]: nodeKey(null) needs to be inside the lambda (failing test)

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java (hunk 3 above)
Comment (diega) [reply ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java:78]: No, we don't really need it, but if I remove that, Sonar complains about an empty block, and if I remove the _try with resources block_ then it complains about not handling the lifecycle of the closable object 🤪

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java (hunk 5 above)
Comment (diega) [reply ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java:116]: yes, I just realized that I have to remove the `.build()` call from the lambda assertion. Fixing it

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java (hunk 1 above)
Comment (diega) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java:91]: super catch! will fix them

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java (hunk 3 above)
Comment (macfarla) [reply ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/network/NetworkingServiceLifecycleTest.java:78]: gotcha. all good

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java (hunk 1 above)
Comment (macfarla) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java:91]: does it make sense to fix those 2 tests in a separate PR?

Comment (diega) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java:91]: I was thinking on adding a first commit with just those changes, but if you think it's better to put it separately to gain more attention I can do it

Comment (macfarla) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java]: A separate commit is probably fine, it's updating tests only. If the new tests fail then that's a different story 😱

Comment (diega) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java]: I had to change the exception type for the test b/c _get_ was throwing `IndexOutOfBoundsException` while _set_ throws `IndexOutOfBoundsException`

Comment (macfarla) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/OperandStackTest.java]: 👀

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**